tests/derive_test.rs

```diff
@@ -309,4 +315,67 @@ fn test_enum_lifetimes() {
             b"\x05\x00",
         ),
     ]);
+
+    assert_roundtrips(&[
+        (Ok(Some(LifetimesChoice::A(17))), b"\x02\x01\x11"),
+        (Ok(Some(LifetimesChoice::B(b"lol"))), b"\x04\x03lol"),
```

Comment:
Other than adding a None in this what makes it different than the first two lines of the assertions directly above?

Reply:
It's testing parsing these as an OPTIONAL, which triggers the `can_parse` pathway